yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala

```diff
@@ -189,6 +190,39 @@ private[spark] abstract class YarnSchedulerBackend(
   }
 
   /**
+   * Renew and update the credentials. This method will trigger credentials renewal in the AM
+   * side, once successfully renewed, it will trigger credentials updating in executor and
+   * driver side.
+   */
+  def updateCredentials(): Unit = {
+    yarnSchedulerEndpoint.amEndpoint match {
+      case Some(am) =>
+        val future = am.ask[Boolean](UpdateCredentials)
+
+        future.onSuccess {
+          case true =>
+            // Update credentials in the executor side only when AM is successfully renewed the
+            // Credentials.
+            synchronized {
+              executorDataMap.values.foreach(_.executorEndpoint.send(UpdateCredentials))
+            }
+            // This will trigger credential updating in the driver side.
+            SparkHadoopUtil.get.triggerCredentialUpdater()
```

Comment:
Note that when we modify the above to return future, this will also need to be in the future (or can always be done inline).